dbms/src/Storages/Transaction/tests/kvstore_helper.h

```
@@ -0,0 +1,154 @@
+extern void CheckRegionForMergeCmd(const raft_cmdpb::AdminResponse & response, const RegionState & region_state);
+extern void ChangeRegionStateRange(RegionState & region_state, bool source_at_left, const RegionState & source_region_state);
+
+namespace tests
+{
+// TODO: Use another way to workaround calling the private methods on KVStore
+class RegionKVStoreTest : public ::testing::Test
+{
+public:
+    RegionKVStoreTest()
+    {
+        test_path = TiFlashTestEnv::getTemporaryPath("/region_kvs_test");
+    }
+
+    static void SetUpTestCase() {}
+
+    void SetUp() override
+    {
+        // clean data and create path pool instance
+        path_pool = createCleanPathPool(test_path);
+
+        reloadKVSFromDisk();
+
+        proxy_instance = std::make_unique<MockRaftStoreProxy>();
+        proxy_helper = std::make_unique<TiFlashRaftProxyHelper>(MockRaftStoreProxy::SetRaftStoreProxyFFIHelper(
+            RaftStoreProxyPtr{proxy_instance.get()}));
+        kvstore->restore(*path_pool, proxy_helper.get());
+        {
+            auto store = metapb::Store{};
+            store.set_id(1234);
+            kvstore->setStore(store);
+            ASSERT_EQ(kvstore->getStoreID(), store.id());
+        }
+    }
+
+    void TearDown() override {}
+
+protected:
+    KVStore & getKVS() { return *kvstore; }
+    KVStore & reloadKVSFromDisk()
+    {
+        kvstore.reset();
+        auto & global_ctx = TiFlashTestEnv::getGlobalContext();
+        kvstore = std::make_unique<KVStore>(global_ctx, TiDB::SnapshotApplyMethod::DTFile_Directory);
+        // only recreate kvstore and restore data from disk, don't recreate proxy instance
+        kvstore->restore(*path_pool, proxy_helper.get());
+        return *kvstore;
+    }
+    void createDefaultRegions()
+    {
+        proxy_instance->init(100);
+    }
+    void initStorages()
+    {
+        registerStorages();
```

we need to try ... catch and ignore the exception thrown by this method. Because other unit test may already call `registerStorages` when we run without any gtest-filter